### PR TITLE
[LTC] correct AdaptiveAvgPool3d channel dim index for shape inference

### LINKED ISSUE #109821: LTC computes wrong shape for nn.AdapativeAvgPool3d
### 🐛 Describe the bug

There is a bug in the shape inference code for 3D adaptive average pooling:

```
import torch
import torch._lazy
import torch_mlir._mlir_libs._REFERENCE_LAZY_BACKEND as lazy_backend

# on CPU device
img_cpu = torch.zeros([2,3,10,10,10])
out_cpu = torch.nn.AdaptiveAvgPool3d(2)(img_cpu)

# on LTC device
lazy_backend._initialize()
device = 'lazy'
img_lazy = img_cpu.to(device)
out_lazy = torch.nn.AdaptiveAvgPool3d(2).to(device)(img_lazy)
torch._lazy.mark_step()

# This check should be True, but it is False
assert out_lazy.numel() == out_cpu.numel(), f"Got {out_lazy.shape} shape on LTC, but {out_cpu.shape} on CPU device"
```

### Versions

master

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @alanwaketan @JackCaoG

### PR DESCRIPTION
**Pull request by davidgens-cerebras**
Fixes #109821

### CODE REVIEW

**Comment by JackCaoG** [comment]
@alanwaketan do you want to review this one?

**Comment by alanwaketan** [comment]
Lol, I think the shape inference code is from https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/AdaptiveAveragePooling.cpp#L44, but that's `adaptive_avg_pool2d_out_cpu_template`.

The right implementation should follow https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/AdaptiveAveragePooling3d.cpp#L87

@davidgens-cerebras Would you mind rewrite the whole logic and add a test case?

**Comment by davidgens-cerebras** [comment]
@alanwaketan Could you point me to the test location you were thinking of? I'm not sure where they live..

**Comment by alanwaketan** [comment]
> @alanwaketan Could you point me to the test location you were thinking of? I'm not sure where they live..

I guess you can add a test case here: https://github.com/pytorch/pytorch/blob/main/test/lazy/test_ts_opinfo.py

**Review by mikaylagawarecki — APPROVED** [review]
Thanks! Could you sign the CLA [in this commen](https://github.com/pytorch/pytorch/pull/109822#issuecomment-1730292233)t please

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Comment by davidgens-cerebras** [comment]
I signed the CLA. The PR is now blocked on a single failing test:
```
win-vs2019-cpu-py3 / build
failed 2 days ago in 2s

Current runner version: '2.309.0'
Runner name: 'i-02fd7bcf8626412b5'
Runner group name: 'Default'
Machine name: 'EC2AMAZ-LPVH52C'
GITHUB_TOKEN Permissions
Secret source: None
Prepare workflow directory
Error: The process cannot access the file 'C:\actions-runner\_work\_actions\pytorch\pytorch\main\test\quantization\core\experimental' because it is being used by another process.
```
Entirely possible I'm missing something, but I don't think that's related to my changes and I seem to be unable to restart this check.. @mikaylagawarecki @alanwaketan could you help with this?

**Comment by alanwaketan** [comment]
@davidgens-cerebras Can you rebase? Hopefully that will resolve the issue.
